UnitTests/ControllerUnitTests/EmployeeControllerTest.cs

```diff
@@ -0,0 +1,293 @@
+            _employeeRepositoryMock.Setup(x => x.UpdateEmployee(employee)).ReturnsAsync((EmployeeModel)null);
+
+            //Act
+            var result = await employeeController.UpdateEmployee(employee);
+
+            //Assert
+            var employeeResult = result.Result as UnprocessableEntityObjectResult;
+            employeeResult?.StatusCode.Should().Be((int)HttpStatusCode.UnprocessableEntity);
+        }
+
+        [Fact]
+        public async Task ShouldThrowInternalServerError_WhenUpdateEmployee_IsCalled()
+        {
+            //Arrange
+            var employeeController = CreateEmployeeController();
+            var employeeId = _fixture.Create<Guid>();
+            var employee = _fixture.Build<EmployeeModel>().With(x => x.EmployeeGuid, employeeId).Create();
+            var exception = _fixture.Create<Exception>();
+            _employeeRepositoryMock.Setup(x => x.GetEmployeeById(employeeId)).ReturnsAsync(employee);
+            _employeeRepositoryMock.Setup(x => x.UpdateEmployee(employee)).Throws(exception);
+
+            //Act
+            var result = await employeeController.UpdateEmployee(employee);
+
+            //Assert
+            var employeeResult = result.Result as StatusCodeResult;
+            employeeResult?.StatusCode.Should().Be((int)HttpStatusCode.InternalServerError);
+        }
+
+        [Fact]
+        public async Task ShouldRemoveEmployee_WhenRemoveEmployee_IsCalled()
+        {
+            //Arrange
+            var employeeController = CreateEmployeeController();
+            var employeeId = _fixture.Create<Guid>();
+            var employee = _fixture.Build<EmployeeModel>().With(x => x.EmployeeGuid, employeeId).Create();
+            _employeeRepositoryMock.Setup(x => x.GetEmployeeById(employeeId)).ReturnsAsync(employee);
+            _employeeRepositoryMock.Setup(x => x.RemoveEmployee(employee)).ReturnsAsync(true);
+
+            //Act
+            var result = await employeeController.RemoveEmployee(employeeId);
+
+            //Assert
+            var employeeResult = result as NoContentResult;
+            employeeResult?.StatusCode.Should().Be((int)HttpStatusCode.NoContent);
+        }
+
+        [Fact]
+        public async Task ShouldReturnNotFound_WhenRemoveEmployee_IsCalled()
+        {
+            //Arrange
+            var employeeController = CreateEmployeeController();
+            var employeeId = _fixture.Create<Guid>();
+            var employee = _fixture.Build<EmployeeModel>().With(x => x.EmployeeGuid, employeeId).Create();
+            _employeeRepositoryMock.Setup(x => x.GetEmployeeById(employeeId)).ReturnsAsync((EmployeeModel)null);
+
+            //Act
+            var result = await employeeController.RemoveEmployee(employeeId);
+
+            //Assert
+            var employeeResult = result as NotFoundResult;
+            employeeResult?.StatusCode.Should().Be((int)HttpStatusCode.NotFound);
```

Comment:
Use the int version of this available as const instead of casting enum

Reply:
using int version of status codes